### PR TITLE
FALCON-1909 Catalog instance triage action fails with null pointer ex…

### PR DESCRIPTION
**Pull request by bvellanki**
…ception.

### CODE REVIEW

File: common/src/main/java/org/apache/falcon/entity/CatalogStorage.java [hunk 1]
```diff
@@ -90,6 +90,7 @@ protected CatalogStorage(Feed feed) throws URISyntaxException {
 
     public CatalogStorage(Cluster cluster, CatalogTable table) throws URISyntaxException {
         this(ClusterHelper.getInterface(cluster, Interfacetype.REGISTRY).getEndpoint(), table);
+        verifyAndUpdateConfiguration(getConf());
```
Comment (yzheng-hortonworks) [inline common/src/main/java/org/apache/falcon/entity/CatalogStorage.java]: You don't need it here. verifyAndUpdateConfiguration is already called in CatalogStorage(String catalogUrl, CatalogTable table).

File: common/src/main/java/org/apache/falcon/entity/CatalogStorage.java [hunk 2]
```diff
@@ -186,13 +193,14 @@ protected CatalogStorage(String uriTemplate) throws URISyntaxException {
...
         this.catalogUrl = uri.getScheme() + "://" + uri.getAuthority();
-
+        verifyAndUpdateConfiguration(getConf());
         parseUriTemplate(uri);
     }
 
     protected CatalogStorage(String uriTemplate, Configuration conf) throws URISyntaxException {
         this(uriTemplate);
         setConf(conf);
+        verifyAndUpdateConfiguration(conf);
```
Comment (yzheng-hortonworks) [inline common/src/main/java/org/apache/falcon/entity/CatalogStorage.java]: You don't need it here. verifyAndUpdateConfiguration is already called in CatalogStorage(String uriTemplate).

Comment (bvellanki) [inline common/src/main/java/org/apache/falcon/entity/CatalogStorage.java]: @yzheng-hortonworks  You are correct in both comments, I was overzealous, will submit a new pull request. Thanks for catching it.

File: common/src/main/java/org/apache/falcon/entity/CatalogStorage.java [hunk 3]
```diff
@@ -100,12 +100,20 @@ protected CatalogStorage(String catalogUrl, String tableUri) throws URISyntaxExc
...
             throw new IllegalArgumentException("Catalog Registry URL cannot be null or empty");
         }
-
+        verifyAndUpdateConfiguration(getConf());
         this.catalogUrl = catalogUrl;
 
         parseFeedUri(tableUri);
     }
 
+    private void verifyAndUpdateConfiguration(Configuration conf) {
```
Comment (yzheng-hortonworks) [inline common/src/main/java/org/apache/falcon/entity/CatalogStorage.java]: Could you move this to be under all the CatalogStorage functions?

Comment (yzheng-hortonworks) [inline common/src/main/java/org/apache/falcon/entity/CatalogStorage.java]: I assume you are always using getConf() as the input argument? Then, a better way is to have verifyAndUpdateConfiguration() with no input and call getConf() in the function, so you don't need to reset configuration with the same value.

**Comment by yzheng-hortonworks** [comment]
+1

**Comment by vrangan** [comment]
+1

**Comment by PraveenAdlakha** [comment]
+1 LGTM

**Pull request closed without merging**